src/components/Form.jsx

```diff
@@ -72,25 +175,33 @@ class Form extends Component {
                 console.log('response', req);
             });
         
-        event.preventDefault();
-        event.reset();
+        //event.preventDefault();
+        //event.reset();
         
         
     }
     
+    errorClass(error) {
+        return(error.length === 0 ? '' : 'has-error');
+    }
+
     render() { 
         return (
             <form className="text-center" id="contactForm" onSubmit={this.handleSubmit} >
+                <div className="panel panel-default">
+                    <FormError formErrors={this.state.formErrors}/>
+                </div>
                 <div className="form-group">
                     <label>
                         Name:
-                        <input id="name" name="clientName" type="text" className="form-control" onChange={this.handleChange}/>
+                        <input required id="name" name="clientName" type="text" className="form-control" onChange={this.handleChange}/>
                     </label>
                 </div>
-                <div>
+                {/*<div className={"form-group" + this.errorClass(this.state.formErrors.clientEmail)}> */}
+                <div className="form-group">
                     <label>
                         Email:
-                        <input id="email" name="clientEmail" type ="text" className="form-control" placeholder="name@example.com" onChange={this.handleChange}/>
+                        <input required id="email" name="clientEmail" type ="text" className="form-control" placeholder="name@example.com" onChange={this.handleChange}/>
```

Comment:
The "required" attribute i added to the html does your length>0 testing for us but your solution allows custom styling as well as full control over the error message and how it is displayed because of this I like your solution better Nice Job!